components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/coordination/task/db/connector/CoordinatedTaskDBConnector.java

```diff
@@ -0,0 +1,362 @@
+
+    /**
+     * Helper method to populate coordinated tasks to the list.
+     *
+     * @param preparedStatement - Statement to be executed to retrieve the list of tasks.
+     * @param taskList          - The list to be populated.
+     * @throws SQLException - Exception.
+     */
+    private void populateCoordinatedTaskList(PreparedStatement preparedStatement, List<CoordinatedTask> taskList)
+            throws SQLException {
+
+        try (ResultSet resultSet = preparedStatement.executeQuery()) {
+            while (resultSet.next()) {
+
+                taskList.add(new CoordinatedTask(resultSet.getString(ConstantUtil.TASK_NAME),
+                                                 resultSet.getString(ConstantUtil.DESTINED_NODE_ID),
+                                                 CoordinatedTask.States
+                                                         .valueOf(resultSet.getString(ConstantUtil.TASK_STATE))));
+            }
+        }
+    }
+
+    /**
+     * Retrieve all the tasks in the task db.
+     *
+     * @return - List of available tasks.
+     */
+    public List<String> getAllTasksInDB() throws SQLException {
+
+        List<String> taskList = new ArrayList<>();
+        try (Connection connection = getConnection(); PreparedStatement preparedStatement = connection.prepareStatement(
+                TaskQueryHelper.RETRIEVE_ALL_TASKS)) {
+            populateCoordinatedTaskNames(preparedStatement, taskList);
+        }
+        debugLogs(taskList);
+        return taskList;
+    }
+
+    /**
+     * Retrieve all assigned and in completed tasks in the task db.
+     *
+     * @return - List of available tasks.
+     */
+    public List<CoordinatedTask> getAllAssignedInCompleteTasks() throws SQLException {
+
+        List<CoordinatedTask> taskList = new ArrayList<>();
+        try (Connection connection = getConnection(); PreparedStatement preparedStatement = connection.prepareStatement(
+                TaskQueryHelper.GET_ALL_ASSIGNED_IN_COMPLETE_TASKS)) {
+            populateCoordinatedTaskList(preparedStatement, taskList);
+        }
+        printDebugLogs(taskList);
+        return taskList;
+    }
+
+    /**
+     * Add the task to the task db.
+     *
+     * @param coordinatedTask - The coordinated task which needs to be added.
+     */
+    public void addTaskToDB(CoordinatedTask coordinatedTask) throws SQLException {
+
+        String taskName = coordinatedTask.getTaskName();
+        CoordinatedTask.States state = coordinatedTask.getTaskState();
+        try (Connection connection = getConnection(); PreparedStatement preparedStatement = connection.prepareStatement(
+                TaskQueryHelper.ADD_TASK)) {
+            preparedStatement.setString(1, taskName);
+            preparedStatement.setString(2, state.name());
+            preparedStatement.executeUpdate();
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Successfully added the task [" + taskName + "] with state [" + state
```

Comment:
We can simply print the task object if we override the toString() method

Reply:
Thanks . Nice suggestion